### ISSUE TITLE
Support Select/Projection of an IQueryable

### ISSUE THREAD

**Issue by pwelter34** [riok/mapperly#214]
**Is your feature request related to a problem? Please describe.**
AutoMapper supports ProjectTo<TDestination>(this IQueryable) for creating a select expression for EF or other querable providers.  Honestly, after this project, its the last reason to use AutoMapper.  If this project could create the needed extension method, that would be super useful.

**Describe the solution you'd like**

Original
```c#
[Mapper]
public static partial class CarMapper
{
    public static partial IQueryable<CarDto> ProjectToCarDto(this IQueryable<Car> source);
}
```

Generated

```c#
public static partial class CarMapper
{
    public static IQueryable<CarDto> ProjectToCarDto(this IQueryable<Car> source)
    {
        return source
            .Select<Car, CarDto>(original =>
                new CarDto
                {
                    Make = original.Make, 
                    Model = original.Model
                }
            );
    }
}
```

**Additional context**

https://docs.automapper.org/en/stable/Queryable-Extensions.html

**Comment by latonz** [comment]
This is something we would like to implement. However, this needs some effort.

Rel. https://github.com/riok/mapperly/discussions/99

**Comment by latonz** [comment]
Idea on how this could work:

A partial method for the projection is defined in the mapper:
```csharp
public static partial IQueryable<CarDto> ProjectToDto(this IQueryable<Car> q);
```

To manually configure properties, additional mapping methods are defined in the mapper as needed:
```csharp
[MapProperty("Model.Name", "ModelName")]
public static partial CarDto ToDto(Car car);
```

Mapperly generates the implementation of the projection method based on the configurations on the other defined methods, if available.
The methods with the configurations are also implemented by Mapperly.
This generates more code than neccessairy,
but it is an approach to make queryable projection mappings configurable.

Complete example:
```csharp
// classes
public class Car
{
    public int Id { set; get; }

    public Manufacturer Manufacturer { set; get; } = new();
}

public class Manufacturer
{
    public string Name { set; get; } = string.Empty;
}

public class CarDto
{
    public int Id { set; get; }
    
    public string Producer { set; get; } = string.Empty;
}

// mapper definition
[Mapper]
public static partial class CarMapper
{
    public static partial IQueryable<CarDto> ProjectToDto(this IQueryable<Car> q);

    [MapProperty("Manufacturer.Name", "Producer")]
    public static partial CarDto ToDto(Car car);
}

// generated mapper implementation
public partial class CarMapper
{
    public static partial IQueryable<CarDto> ProjectToDto(IQueryable<Car> q)
    {
        return System.Linq.Queryable.Select(q, x => new CarDto() { Id = x.Id, Producer = x.Manufacturer.Name });
    }

    public static partial CarDto ToDto(Car car)
    {
        var target = new CarDto();
        target.Id = car.Id;
        target.Producer = x.Manufacturer.Name;
        return target;
    }
}
```

**Comment by latonz** [comment]
A first preview of this is released in `2.8.0-next.1`. Feel free to give it a try (docs available [here](https://next.mapperly.pages.dev/docs/configuration/queryable-projections)). Any feedback is welcome 😊
